src/App.tsx

```diff
@@ -47,49 +61,123 @@ export function getReorderedGoods(
 //   sortType: SortType,
 // };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, ReorderOptions> {
+  state: Readonly<ReorderOptions> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  clickOnSort = (sortType: SortType) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        this.setState({
+          sortType: SortType.ALPHABET,
+        });
+
+        break;
+
+      case SortType.LENGTH:
+        this.setState({
+          sortType: SortType.LENGTH,
+        });
+
+        break;
+
+      case SortType.NONE:
+        this.setState({
+          sortType: SortType.NONE,
+        });
+
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  clickOnReverse = () => (
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }))
+  );
+
+  clickOnReset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const isResetVisible = sortType !== SortType.NONE || isReversed;
+
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={cn(
+              'button',
+              'is-info',
+              {
+                'is-light': sortType !== SortType.ALPHABET,
+              },
+            )}
```

Comment:
you want to have it maximally concise
```suggestion
            className={cn(button is-info', {
                'is-light': sortType !== SortType.ALPHABET,
              })}
```